modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -243,6 +254,40 @@ private void applyLifecycle(List<String> createdComponentModels, boolean applySt
     });
   }
 
+  private BeanDependencyResolver newBeanDependencyResolver() {
```

Comment:
Does this change for every execution? Wouldn't it be better to have it as a class attribute?

Reply:
Yeap, nice catch.